manifests/install.pp

```diff
@@ -14,9 +14,6 @@
       ensure => "directory";
   }
 
-  systemd::unit_file {"${nats::service_name}.service":
-    content => epp("nats/systemd_service.epp")
-  }
-
+  Class["nats::install::${nats::service_type}"] ~> Class["nats::service"]
```

Comment:
I wasn't sure if the service would get notified if the contents of the init script or Upstart job changed this way, so I added the extra dependency chain. Will test that again in my Vagrant setup.